wye/profiles/forms.py

```diff
@@ -14,25 +16,45 @@ class UserAuthenticationForm(AuthenticationForm):
 
 
 class SignupForm(forms.ModelForm):
-    mobile = forms.CharField(label=_("Mobile"), max_length=10,
-                             required=True,
-                             widget=forms.TextInput(attrs={'placeholder': 'Mobile Number'}))
+    mobile = forms.CharField(
+        label=_("Mobile"),
+        max_length=10,
+        widget=forms.TextInput(
+            attrs={'placeholder': 'Mobile'}
+        )
+    )
+    usertype = forms.ModelMultipleChoiceField(
+        queryset=models.UserType.objects.exclude(slug='admin'))
+    location = forms.ModelChoiceField(queryset=Location.objects.all())
+    interested_locations = forms.ModelMultipleChoiceField(
+        queryset=Location.objects.all())
+    interested_sections = forms.ModelMultipleChoiceField(
+        queryset=WorkshopSections.objects.all())
 
     class Meta:
         model = get_user_model()
         fields = ['first_name', 'last_name']
         widgets = {
-            'first_name': forms.TextInput(attrs={'placeholder': 'First Name'}),
+            'first_name': forms.TextInput(attrs={'placeholder': 'First Name', 'autofocus': 'on'}),
             'last_name': forms.TextInput(attrs={'placeholder': 'Last Name'}),
         }
 
-    def save(self, user):
-        user.profile.mobile = self.cleaned_data['mobile']
+    def signup(self, request, user):
```

Comment:
Well Because django-allauth doc suggest to use signup method when you have another signup class from 'ACCOUNT_SIGNUP_FORM_CLASS' for saving user.
```
ACCOUNT_SIGNUP_FORM_CLASS (=None) A string pointing to a custom form class (e.g.
‘myapp.forms.SignupForm’) that is used during signup to ask the user for additional input (e.g. newslet-
ter signup, birth date). This class should implement a def signup(self, request, user) method, where user
represents the newly signed up user
```
[Source](https://django-allauth.readthedocs.org/en/latest/configuration.html)